gobblin-utility/src/main/java/org/apache/gobblin/util/hadoop/TokenUtils.java

```diff
@@ -193,9 +192,12 @@ public static void getHiveToken(final State state, IMetaStoreClient hiveClient,
     try {
       // Fetch and save the default hcat token.
       LOG.info("Fetching default Hive MetaStore token from hive");
-      HiveConf hiveConf = new HiveConf();
 
-      Token<DelegationTokenIdentifier> hcatToken = fetchHcatToken(userToProxy, hiveConf, null, hiveClient);
+      // Fetch and save the default hcat token.
+      // The tokenSignature needs to be matched with the one in HiveConf to correctly identity a specific token.
```

Comment:
Might be better to restate the comment, something along these lines:
Fetch the delegation token with "service" field overwritten with the metastore.uri configuration. \<link to the factory code\> sets the signature field to the same value to retrieve the token correctly.

Reply:
Thanks for the advice.